### PR TITLE
Refactored Button component

### PR DESCRIPTION
**Pull request by chrisvanmook**
- Added: react-docgen-typescript-loader to storybook config for showing props with description

- Changed: Made IconButton and ShareBar deprecated. Also refactored this in other components than
  use IconButton
- Added: stories for buttons
- Changed: Refactored the Button component to use the prop "variants" instead of "color".
- Removed: ShareBar, named export however still exists

## Amsterdam styled components pull request

Before opening a pull request, please ensure:

- [x] The default export from a file matches the file name
- [x] You have been following the guidelines, written in the [README](../docs/CONTRIBUTING.md#user-content-conventions-and-rules) file
- [x] Your code has the necessary tests written
- [x] You have updated the [CHANGELOG.md unreleased sections](../CHANGELOG.md#muser-content-unreleased)

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: packages/asc-ui/src/components/Button/ButtonStyle.ts [hunk 1]
```diff
@@ -1,41 +1,177 @@
...
+  color?: Theme.TypeLevel
+  /**
+   * Deprecated: use size to create a button with equal width and height
+   * @deprecated
+   */
+  square?: boolean
+  /**
+   * Set an equal height and width
+   */
+  size?: number
```
Comment (royderks) [inline packages/asc-ui/src/components/Button/ButtonStyle.ts]: How can you set an equal height and width with this?

Comment (chrisvanmook) [reply packages/asc-ui/src/components/Button/ButtonStyle.ts]: By using the [size](https://github.com/Amsterdam/amsterdam-styled-components/pull/91/files#diff-317386189eb84716077d89eb245d5142R150) util from polished. Will rewrite this to height: size, width: size when we decide to really get rid of polished

File: packages/asc-ui/src/components/Button/ButtonStyle.ts [hunk 2]
```diff
@@ -1,41 +1,182 @@
...
+  font-weight: 500;
+  line-height: 20px;
+  padding: ${({ size: sizeProp }) => (sizeProp ? '0' : '12px 15px')};
+  ${({ size: sizeProp, square, hasIconLeft, hasIconRight }) =>
+    sizeProp || square // make the button square
+      ? css`
+          display: flex;
+          align-items: center;
+          justify-content: center;
+          ${size(sizeProp || 30)} // width and height
```
Comment (royderks) [inline packages/asc-ui/src/components/Button/ButtonStyle.ts]: use default prop from the component?

File: packages/asc-ui/src/components/Button/ButtonStyle.ts [hunk 3]
```diff
@@ -1,41 +1,182 @@
...
+        color: ${getColor('primary')({ theme })};
+        border: 1px solid ${getColor('primary')({ theme })};
+        ${svgFill('primary')};
+
+        &:hover {
+          outline: 1px solid ${getColor('primary')({ theme })};
+        }
+      `
+
+    case 'blank':
```
Comment (aditudorache) [inline packages/asc-ui/src/components/Button/ButtonStyle.ts]: can pu please add a short description for this one? the rest of variants are clear wat they do.

**Pull request merged**